### DISCUSSION TITLE
Request for Feedback on My GitHub Repository

### DISCUSSION THREAD

**Discussion by pisoc912** [community/community#118492] [Programming Help]
### Select Topic Area

Bug

### Body

Hello everyone,

I recently completed a take-home task as part of a job application process, and I received feedback that my submission contained bugs and was not well-structured. As I am continuously looking to improve my skills and learn from my experiences, I am reaching out to the community for constructive feedback.

Here is the link to my project on GitHub: [[insert link to the repository]](https://github.com/pisoc912/mapbox)

I specifically worked on [briefly describe what the task was about, what technologies you used, and any particular challenges you encountered].

I would be very grateful if experienced frontend engineers could take a look at my code and provide feedback on:

Identifying any bugs or issues that might be present
Suggestions for improving the code structure and organization
Best practices that I could implement to enhance the quality of the code
Any other aspects that you think could improve the overall project
I am open to any constructive criticism and suggestions that could help me grow and refine my skills as a frontend developer.

Thank you very much for taking the time to review my work and for any insights you can provide.

Best regards,

**Comment by Pranav-Karra-3301** [comment]
Hello,

I've had the opportunity to review your project on GitHub, and I must say, I found it quite interesting. Your README file was particularly well-organized and easy to understand, which made navigating through your project a breeze. The clarity of your documentation certainly sets a strong foundation for collaboration and further development.

It's apparent that you've put a significant amount of effort into this project. If you have any specific areas you would like feedback on or any questions about potential improvements, feel free to reach out!

Best wishes on your coding journey.

Kind regards
Pranav Karra

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @pisoc912 !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!

P.S. I second what @Pranav-Karra-3301 wrote. Your fundamentals are on point! 
As for the packages you are using; I'm not familiar with them; but I can offer some house keeping suggestions.

I'm not sure what the industry at large is switching to (from Create-React-App), my circles started using Vite. Though this being mainly a component, it may not matter much. I think the largest compatibility issue would be the use of .jsx v. .js.

I noticed some Chinese characters in your comments: mapService.js line 163.  

Great work so far!!!
